lptree.pas

```diff
@@ -4174,6 +4186,104 @@ constructor TLapeTree_InternalMethod_Raise.Create(ACompiler: TLapeCompilerBase;
   FForceParam := True;
 end;
 
+function TLapeTree_InternalMethod_Find.resType: TLapeType;
+begin
+  if (FResType = nil) then
+    FResType := FCompiler.getBaseType(ltInt32);
+
+  Result := inherited;
+end;
+
+function TLapeTree_InternalMethod_Find.Compile(var Offset: Integer): TResVar;
+var
+  _Find: TLapeGlobalVar;
+  Item, Arr: TResVar;
+begin
+  Result := NullResVar;
+  Dest := NullResVar;
+
+  if (FParams.Count <> 2) or isEmpty(FParams[0]) or isEmpty(FParams[1]) then
+    LapeExceptionFmt(lpeWrongNumberParams, [2], DocPos);
+
+  Item := FParams[0].Compile(Offset);
+  Arr := FParams[1].Compile(Offset);
+
+  if (Item.VarType.BaseType in LapeArrayTypes - LapeStringTypes) then // Lape does not have array comparison
+    LapeException(lpeInvalidEvaluation, DocPos);
```

Comment:
Some more fancy error messages maybe? Don't want to make things more cryptic as they already are :)